src/ride/station.c

```diff
@@ -64,7 +64,10 @@ static void ride_update_station_blocksection(rct_ride *ride, int stationIndex)
 {
 	rct_map_element *mapElement;
 
+	assert(ride != NULL);
 	mapElement = ride_get_station_start_track_element(ride, stationIndex);
+	assert(mapElement != NULL);
+
```

Comment:
It would be better to return instead of this assert and have a warning since all rides should have a station